include/mm/pmm.h

```diff
@@ -76,6 +76,13 @@ extern frame_t *get_free_frames(unsigned int order);
 extern void put_free_frames(mfn_t mfn, unsigned int order);
 extern void reclaim_frame(mfn_t mfn, unsigned int order);
 
+extern frame_t *find_free_mfn_frame(mfn_t mfn, unsigned int order);
+extern frame_t *find_busy_mfn_frame(mfn_t mfn, unsigned int order);
+extern frame_t *find_mfn_frame(mfn_t mfn, unsigned int order);
+extern frame_t *find_free_paddr_frame(paddr_t paddr);
+extern frame_t *find_busy_paddr_frame(paddr_t paddr);
+extern frame_t *find_paddr_frame(paddr_t paddr);
```

Comment:
I think either we should use the free,busy sugar functions, or just not include them at all?

Reply:
For now, with the current PMM implementation, the split for free/busy and any (`find_mfn_frame`) is a small optimization as main current needs and uses are very specific (and for busy frames), but general API has been created too.

Reply:
Okey. I think I didn't think of them as API for tests that we write.

This is pretty nice, that we can allocate guest-physical memory easily. And the host physical base mapping can actually a small qemu wrapper manage :D